lib/apytypes/_array_functions.py

```diff
@@ -411,6 +411,361 @@ def expand_dims(a, axis):
+    """
+    try:
+        eye = a_type.eye
+    except AttributeError:
+        raise TypeError(f"Cannot make eye array of type {type(a_type)}")
+    if hasattr(a_type, "int_bits") and hasattr(a_type, "frac_bits"):
+        return eye(n=n, m=m, int_bits=int_bits, frac_bits=frac_bits)
+
+    if hasattr(a_type, "exp_bits") and hasattr(a_type, "mantissa_bits"):
+        return None
+
+    raise ValueError("Only 'fixed' and 'float' array_types are defined")
+
+
+def identity(a_type, n, int_bits=8, frac_bits=8, exp_bits=None, mantissa_bits=None):
+    """
+    Return the identity array.
+
+    The identity array is a square array with ones on the main diagonal.
+
+    Parameters
+    ----------
+    a_type : :class:`APyFloatArray` or :class:`APyFixedArray`, optional
+        The type of array to initialize.
+    n : int
+        Number of rows (and columns) in n x n output.
+    int_bits : int, optional
+        Number of integer bits.
+    frac_bits : int, optional
+        Number of fractional bits.
+    exp_bits : int, optional
+        Number of exponential bits.
+    mantissa_bits : int, optional
+        Number of mantissa bits.
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        n x n array with its main diagonal set to one, and all other elements 0.
+    """
+    try:
+        identity = a_type.identity
+    except AttributeError:
+        raise TypeError(f"Cannot make identity array of type {type(a_type)}")
+
+    if hasattr(a_type, "int_bits") and hasattr(a_type, "frac_bits"):
+        return identity(n=n, int_bits=int_bits, frac_bits=frac_bits)
+
+    if hasattr(a_type, "exp_bits") and hasattr(a_type, "mantissa_bits"):
+        return None
+
+    raise ValueError("Only 'fixed' and 'float' array_types are defined")
+
+
+def full(
+    a_type,
+    shape,
+    fill_value,
+    int_bits=None,
+    frac_bits=None,
+    exp_bits=None,
+    mantissa_bits=None,
+):
+    """
+    Return a new array of given shape and type, filled with fill_value.
+
+    Parameters
+    ----------
+    a_type : :class:`APyFloatArray` or :class:`APyFixedArray`, optional
+        The type of array to initialize.
+    shape : tuple
+        Shape of the array.
+    fill_value : :class:`APyFloat` or :class:`APyFixedArray` or int or float
+        Fill value.
+    int_bits : int, optional
+        Number of integer bits.
+    frac_bits : int, optional
+        Number of fractional bits.
+    exp_bits : int, optional
+        Number of exponential bits.
+    mantissa_bits : int, optional
+        Number of mantissa bits.
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        Array of fill_value with the given shape and a_type.
+    """
+    try:
+        full = a_type.full
+    except AttributeError:
+        raise TypeError(f"Cannot make full array of type {type(a_type)}")
+
+    if hasattr(a_type, "int_bits") and hasattr(a_type, "frac_bits"):
```

Comment:
I think it is probably better to check for the type explicitly (although I do not know if `a_type == APyFixedArray` or `a_type is APyFixedArray` works...). Even though I understand the idea of handling a more general class of arrays.

Reply:
I did try 
if isinstance(a_type, APyFixedArray) but didn't work because but for some reason the type(a_type) was something along the lines of  'nanobind object 0x000001'. Although, I didn't look into it that deep. Does this ring any bells?

Reply:
I wrote `isinstance` first, but the realized that `a_type` is not an instance of `APyFixedArray`, but the class itself. Hence, `==` or `is`.